src/data/fintype/basic.lean

```diff
@@ -1390,6 +1390,59 @@ lemma finset.univ_map_embedding {α : Type*} [fintype α] (e : α ↪ α) :
   univ.map e = univ :=
 by rw [←e.equiv_of_fintype_self_embedding_to_embedding, univ_map_equiv_to_embedding]
 
+/-- There exists a bijection between two types of the same finite cardinality, which extends a given
+bijection on a given finset. -/
+lemma finset.exists_equiv_extend_of_card_eq [fintype α] [decidable_eq β] {t : finset β}
+  (hαt : fintype.card α = t.card) {s : finset α} :
+  ∀ {f : α → β} (hfst : s.image f ⊆ t) (hfs : set.inj_on f s),
+    ∃ g : α ≃ t, ∀ i ∈ s, (g i : β) = f i :=
```

Comment:
You could generalize from `fintype α` to a second `finset α` like
```suggestion
lemma finset.exists_equiv_extend_of_card_eq [decidable_eq β] {s s' : finset α} {t : finset β}
  (hst : s'.card = t.card) (hs : s ⊆ s') :
  ∀ {f : α → β} (hfst : s.image f ⊆ t) (hfs : set.inj_on f s),
    ∃ g : s' ≃ t, ∀ i (hi : i ∈ s), (g ⟨i, hs hi⟩  : β) = f i :=
```

Reply:
This is less useful, I think, because then in the typical use case one needs a further identification between `α` and `↥finset.univ`.